### PR TITLE
[REVIEW] upgrade to RAPIDS 0.11

### PR DESCRIPTION
**Pull request by yidong72**
Since RAPIDS 0.11 is released, move the changes from [PR](https://github.com/rapidsai/gQuant/pull/65) to this PR. I didn't include 
```
RUN echo "conda 4.7.12" > /opt/conda/conda-meta/pinned \
    && conda update -n base -c defaults conda -y
```
Not convinced it solved the problem.

### CODE REVIEW

**Comment by miguelangel** [comment]
Hi Yi,

The new RAPIDS container (v0.11) already includes conda v4.7.12, so you are right, that change is not needed anymore.

Thank you for taking care of this PR.

Regards,
Miguel

**Comment by yidong72** [comment]
> Hi Yi,
> 
> The new RAPIDS container (v0.11) already includes conda v4.7.12, so you are right, that change is not needed anymore.
> 
> Thank you for taking care of this PR.
> 
> Regards,
> Miguel

Thanks for the clarification of this.

**Comment by yidong72** [comment]
Added the profile argument to help optimize the performance. Now we can evaluate the graph by:
```python
o_gpu = task_graph.run(
            outputs=outlist,
            replace=replace_spec, profile=True)
```

File: gquant/dataframe_flow/_node_flow.py [hunk 1]
```diff
@@ -672,6 +672,20 @@ def get_pout(out_dict, port):
 
         return output_df
 
+    def decorate_process(self):
+        import time
+        def timer(*argv):
+            start = time.time()
+            result = self.process(*argv)
+            end = time.time()
+            print('id:%s process time:%.3f' % (self.uid, end-start))
```
[the remark this answers is not in the document]
Comment (yidong72) [reply gquant/dataframe_flow/_node_flow.py]: done

File: docker/build.sh [hunk 2]
```diff
@@ -2,47 +2,59 @@
...
+	;;
 esac
 
-CONTAINER="nvcr.io/nvidia/rapidsai/rapidsai:0.10-cuda${CONTAINER_VER}-runtime-ubuntu${OS_STR}"
-
+CONTAINER="nvcr.io/nvidia/rapidsai/rapidsai:0.11-cuda${CONTAINER_VER}-runtime-${OS_STR}"
 
 D_FILE=${D_FILE:='Dockerfile.Rapids'}
-D_CONT=${D_CONT:='gquant/gquant:latest'}
+D_CONT=${D_CONT:="gquant/gquant:${OS_STR}"}
```
[the remark this answers is not in the document]
Comment (yidong72) [reply docker/build.sh]: Good suggestions

**Comment by yidong72** [comment]
I did all the changes as you suggested. please review it again.

File: docker/build.sh [hunk 3]
```diff
@@ -52,16 +65,20 @@ cp ../setup.py ./gQuant
 cp ../LICENSE ./gQuant
 rsync -av --progress ../notebooks ./gQuant --exclude data --exclude .cache --exclude many-small --exclude storage --exclude dask-worker-space --exclude __pycache__
 
+gquant_ver=$(grep version gQuant/setup.py | sed "s/^.*version='\([^;]*\)'.*/\1/")
+D_CONT=${D_CONT:="gquant/gquant:${gquant_ver}_${OS_STR}_${CUDA_VERSION}_${RAPIDS_VERSION}"}
```
Comment (avolkov1) [inline docker/build.sh]: This version string looks a bit off. I built for Ubuntu 18 with CUDA 10.1.2 and version came out like this:
```
gquant/gquant:0.3_ubuntu18.04_3_0.11
```
The `CUDA_VERSION` variable isn't actually the cuda version, it's the option for which cuda version to use. The cuda version is indicated by the `CONTAINER_VER`. It should look like this:
```
D_CONT=${D_CONT:="gquant/gquant:${gquant_ver}_${OS_STR}_cuda${CONTAINER_VER}_rapids${RAPIDS_VERSION}"}
```
Which will produce:
```
gquant/gquant:0.3_ubuntu18.04_cuda10.1_rapids0.11
```

Comment (yidong72) [reply docker/build.sh]: fixed

**Comment by yidong72** [comment]
the CUDA version tag is fixed

**Pull request merged**